src/validation.cpp

```diff
@@ -3115,7 +3115,7 @@ static bool AcceptBlock(const std::shared_ptr<const CBlock>& pblock, CValidation
     // process an unrequested block if it's new and has enough work to
     // advance our tip, and isn't too many blocks ahead.
     bool fAlreadyHave = pindex->nStatus & BLOCK_HAVE_DATA;
-    bool fHasMoreWork = (chainActive.Tip() ? pindex->nChainWork > chainActive.Tip()->nChainWork : true);
+    bool fHasMoreOrSameWork = (chainActive.Tip() ? pindex->nChainWork >= chainActive.Tip()->nChainWork : true);
```

Comment:
In commit "Accept unrequested blocks with work equal to our tip"

Same question again: Is this change related to the "new headers are not a descendant of an invalid block" check or is it only part of this PR because it simplifies the p2p-acceptblock test? Again it seems fine, just asking to understand.

Reply:
The second. Mostly because it started as a "cleanup p2p-acceptblock test" pr and then I didn't want to rewrite the p2p-acceptblock test again to not care about this.